internal/service/resources/collector/collector_hwplugin.go

```diff
@@ -199,15 +199,14 @@ func (d *HwPluginDataSource) convertResourcePool(pool *inventoryclient.ResourceP
 	}
 }
 
-// MakeResourceID calculates a UUID value to be used as the ResourceID.  The cloudID and hwPluginRef are added to the node
-// id value to ensure we get a globally unique value.
-func MakeResourceID(cloudID uuid.UUID, hwPluginRef, hwMgrNodeID string) uuid.UUID {
-	return ctlrutils.MakeUUIDFromNames(ResourceUUIDNamespace, cloudID, hwPluginRef, hwMgrNodeID)
-}
-
 func (d *HwPluginDataSource) convertResource(resource *inventoryclient.ResourceInfo) *models.Resource {
-	// The resourceID computed here must
-	resourceID := MakeResourceID(d.cloudID, d.hwplugin.Name, resource.ResourceId)
+	// Parse the resource ID directly (now it's the BMH UID from the hardware plugin)
+	resourceID, err := uuid.Parse(resource.ResourceId)
+	if err != nil {
+		slog.Warn("Failed to parse resource ID as UUID, using nil UUID", "resourceId", resource.ResourceId, "error", err)
```

Comment:
this should be an error.  not being able to identify the resource by its id value is critical.

Reply:
I agree it should never happen, but we should error out here if it does.  Sending bad data up to the client is not going to be helpful.